pkgs/native_assets_builder/lib/src/build_runner/build_runner.dart

```diff
@@ -542,7 +595,113 @@ ${e.message}
     }
   }
 
+  /// Compiles the hook to dill and caches the dill.
+  ///
+  /// It does not reuse the cached dill for different [config]s, due to
+  /// reentrancy requirements. For more info see:
+  /// https://github.com/dart-lang/native/issues/1319
+  Future<(bool success, File kernelFile, DateTime lastSourceChange)>
+      _compileHookForPackageCached(
+    HookConfigImpl config,
+    Uri packageConfigUri,
+    Uri workingDirectory,
+    bool includeParentEnvironment,
+  ) async {
+    final kernelFile = File.fromUri(
+      config.outputDirectory.resolve('../hook.dill'),
+    );
+    final depFile = File.fromUri(
+      config.outputDirectory.resolve('../hook.dill.d'),
+    );
+    final bool mustCompile;
+    final DateTime sourceLastChange;
+    if (!await depFile.exists()) {
+      mustCompile = true;
+      sourceLastChange = DateTime.now();
+    } else {
+      final depFileContents = await depFile.readAsString();
+      final dartSourceFiles = depFileContents
```

Comment:
Maybe add a reference to/example of a depFile, so that this parsing logic can be understood?